edge/pkg/edged/edged_pods.go

```diff
@@ -754,6 +790,30 @@ func (e *edged) makeEnvironmentVariables(pod *v1.Pod, container *v1.Container, p
 					}
 					return result, fmt.Errorf("couldn't find key %v in ConfigMap %v/%v", key, pod.Namespace, name)
 				}
+			case envVar.ValueFrom.SecretKeyRef != nil:
+				sc := envVar.ValueFrom.SecretKeyRef
+				name := sc.Name
+				key := sc.Key
+				optional := sc.Optional != nil && *sc.Optional
+				secret, ok := secrets[name]
+				if !ok {
+					if e.kubeClient == nil {
+						return result, fmt.Errorf("couldn't get secret %v/%v, no kubeClient defined", pod.Namespace, name)
+					}
+					secret, err = e.metaClient.Secrets(pod.Namespace).Get(name)
+					if err != nil {
+						if apierrors.IsNotFound(err) && optional {
+							// ignore error when marked optional
+							continue
+						}
+						return result, err
+					}
+					secrets[name] = secret
+				}
+				runtimeVal = string(secret.Data[key])
+				if optional {
+					continue
+				}
```

Comment:
```suggestion
				runtimeValBytes, ok := secret.Data[key]
				if !ok {
					if optional {
						continue
					}
					return result, fmt.Errorf("couldn't find key %v in Secret %v/%v", key, pod.Namespace, name)
				}
				runtimeVal = string(runtimeValBytes)
```
Let's keep same with K8s.